subgraphs/abracadabra/src/positions.ts

```diff
@@ -388,3 +202,234 @@ function getAccountBalance(
 
   return tryBalance.reverted ? BIGINT_ZERO : tryBalance.value;
 }
+
+export function getLiquidatePosition(
+  side: string,
+  marketId: string,
+  accountId: string
+): string {
+  const positionCounter = PositionCounter.load(
+    accountId.concat("-").concat(marketId).concat("-").concat(side)
+  );
+  if (!positionCounter) {
+    log.warning("No liquidation position found for account {} on market {}", [
+      accountId,
+      marketId,
+    ]);
+    return "";
+  }
+
+  const position = Position.load(
+    positionCounter.id.concat("-").concat(positionCounter.nextCount.toString())
+  );
+  if (!position) {
+    log.warning("No liquidation position found for account {} on market {}", [
+      accountId,
+      marketId,
+    ]);
+    return "";
+  }
+
+  return position.id;
+}
+
+// A series of side effects on position added
+// They include:
+// * Create a new position when needed or reuse the exisitng position
+// * Update position related data in protocol, market, account
+// * Take position snapshot
+function addPosition(
+  protocol: LendingProtocol,
+  market: Market,
+  account: Account,
+  newBalance: BigInt,
+  eventType: string,
+  side: string,
+  event: ethereum.Event
+): Position {
+  const counterID = account.id
+    .concat("-")
+    .concat(market.id)
+    .concat("-")
+    .concat(side);
+  let positionCounter = PositionCounter.load(counterID);
+  if (!positionCounter) {
+    positionCounter = new PositionCounter(counterID);
+    positionCounter.nextCount = 0;
+    positionCounter.save();
+  }
+  const positionID = positionCounter.id
+    .concat("-")
+    .concat(positionCounter.nextCount.toString());
+
+  let position = Position.load(positionID);
+  const openPosition = position == null;
+  if (openPosition) {
```

Comment:
Is there a reason or preference to do this instead of just `if (!position)`?

Reply:
Oh, I didn’t mean to refactor it into the opposite case to do an early return. Was asking about 
```
const openPosition = position == null;
if (openPosition) {
 ...
}

// instead of just:
if (!position) { … }

```

Just curious about the redundancy. Is it more legible? Or feels safer when you know you are using a boolean? or? :D

Reply:
On the other instance you suggested we do an early return lmao, so that is what i did in the v3 lending sdk... I realized it was nicer looking to have the double conditions, now there are like 8 lines of duplicate code, but not a big deal

Either way we have to update code in 2 diff places if something happens 😵‍💫🤷‍♂️